system_setup/common/wsl_conf.py

```diff
@@ -143,18 +146,25 @@ def drop_if_exists(self, config_section, config_setting):
                 self.config.write(configfile)
 
     def update(self, config_section, config_setting, config_value):
+        log.debug("Updating config: %s %s %s",
+                  config_section, config_setting, config_value)
+        if not self.config.has_section(config_section):
+            self.config.add_section(config_section)
+        if isinstance(config_value, bool):
+            config_value = str(config_value).lower()
         self.config[config_section][config_setting] = config_value
         with open(self.conf_file, 'w') as configfile:
             self.config.write(configfile)
 
 
 class WSLConfigHandler:
 
-    def __init__(self, is_dry_run):
+    def __init__(self, root_dir, is_dry_run):
+        self.root = root_dir
         self.is_dry_run = is_dry_run
         self.ubuntu_conf = \
-            WSLConfig("/etc/ubuntu-wsl.conf")
-        self.wsl_conf = WSLConfig("/etc/wsl.conf")
+            WSLConfig(self.root + "/etc/ubuntu-wsl.conf")
```

Comment:
I think dry-run mode should directly override the whole path. This means that by default in dry-run, you would end up with `.subiquity/wsl.conf` for instance, as this is where all the other assets are generated in dry-run mode.